### PR TITLE
show different ndef records

### PR DESCRIPTION
**Pull request by himalia416**

### CODE REVIEW

File: profile_nfc/build.gradle.kts [hunk 1]
```diff
@@ -17,10 +17,14 @@ dependencies {
...
+    implementation(libs.nordic.core)
 
     implementation(project(":settings"))
     implementation(project(":domain"))
     implementation(project(":remoteDatabase"))
     implementation(project(":settingsStorage"))
     implementation(project(":welcome"))
+
+    // Balloon for tooltip popup.
+    implementation ("com.github.skydoves:balloon-compose:1.5.2")
```
Comment (sylwester-zielinski) [inline profile_nfc/build.gradle.kts:29]: Add to toml.

File: profile_nfc/src/main/java/no/nordicsemi/profile_nfc/viewmodel/NfcViewModel.kt [hunk 2]
```diff
@@ -43,15 +44,28 @@ internal class NfcViewModel @Inject constructor(
...
+                    when {
+                        !it.isNfcSupported -> showNfcNotSupported()
+                        !it.isNfcEnabled -> showNfcNotEnabledPage()
+                        it.tag != null -> tagDiscovered()
+                        else -> showScanTag()
+                    }
+                }.launchIn(viewModelScope)
             }
-        }.launchIn(viewModelScope)
+            .apply {
```
Comment (sylwester-zielinski) [inline profile_nfc/src/main/java/no/nordicsemi/profile_nfc/viewmodel/NfcViewModel.kt:58]: But apply after apply is for sure an overkill

File: app/src/main/java/no/nordicsemi/nfcapplication/MainActivity.kt [hunk 3]
```diff
@@ -1,35 +1,43 @@
...
         setContent {
             NordicTheme {
-                NavigationView(destinations = NfcDestinations)
+                val viewModel = hiltViewModel<NfcScanningManagerVM>()
+                val activity = LocalContext.current as Activity
+                DisposableEffect(key1 = viewModel) {
+                    viewModel.onResume(activity)
+                    onDispose { viewModel.onPause(activity) }
+                }
+                SelectionContainer {
```
Comment (sylwester-zielinski) [inline app/src/main/java/no/nordicsemi/nfcapplication/MainActivity.kt:38]: Are you sure it should be on everything?

File: navigation/src/main/java/no/nordicsemi/navigation/NfcNavitagation.kt [hunk 4]
```diff
@@ -2,9 +2,13 @@ package no.nordicsemi.navigation
```
Comment (sylwester-zielinski) [inline navigation/src/main/java/no/nordicsemi/navigation/NfcNavitagation.kt:2]: Fix file name

File: nfcScanner/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+    implementation(project(":domain"))
+    implementation(project(":remoteDatabase"))
+    implementation(project(":settingsStorage"))
+    implementation(project(":welcome"))
+    implementation(project(":nfcUi"))
+    implementation(project(":settings"))
+
+    // Balloon for tooltip popup.
+    implementation ("com.github.skydoves:balloon-compose:1.5.2")
```
Comment (sylwester-zielinski) [inline nfcScanner/build.gradle.kts:31]: toml

File: nfcScanner/src/main/java/no/nordicsemi/nfcscanner/views/NfcScanningScreen.kt [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+@OptIn(ExperimentalMaterial3Api::class)
+@RequiresApi(Build.VERSION_CODES.S)
+@Composable
+fun NfcScanningScreen() {
+    RequireNfc {
+        val nfcViewModel: NfcScanningViewModel = hiltViewModel()
+        val nfcState by nfcViewModel.state.collectAsState()
+
+        when (nfcState.setting?.showWelcomeScreen) {
+            true -> {
```
Comment (sylwester-zielinski) [inline nfcScanner/src/main/java/no/nordicsemi/nfcscanner/views/NfcScanningScreen.kt:31]: I think if is better for this case

File: nfcScanner/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="app_name">NFC Application</string>
+
+    <string name="scan_nfc_tag">Scan an NFC tag…</string>
+    <string name="nfc_not_enabled_appbar">NFC not Enabled</string>
+    <string name="nfc_not_enabled">NFC not Enabled</string>
```
Comment (sylwester-zielinski) [inline nfcScanner/src/main/res/values/strings.xml:7]: Do you need this? Haven't you moved this to commons?

File: nfcUi/build.gradle.kts [hunk 8]
```diff
@@ -17,10 +17,10 @@ dependencies {
     implementation(libs.androidx.compose.material.iconsExtended)
     implementation(libs.androidx.compose.bom)
     implementation(libs.androidx.hilt.navigation.compose)
+    implementation(libs.nordic.core)
 
-    implementation(project(":settings"))
+    // Balloon for tooltip popup.
+    implementation ("com.github.skydoves:balloon-compose:1.5.2")
```
Comment (sylwester-zielinski) [inline nfcUi/build.gradle.kts:23]: toml

**Review by sylwester-zielinski — COMMENTED** [review]
Have you checked all my previous comments?

File: nfcScanner/src/main/res/values/strings.xml (hunk 7 above)
Comment (himalia416) [reply nfcScanner/src/main/res/values/strings.xml]: fixed

**Pull request merged**